src/core/infer_response.cc

```diff
@@ -94,6 +97,22 @@ InferenceResponse::AddOutput(
   return Status::Success;
 }
 
+Status
+InferenceResponse::ClassificationLabel(
+    const InferenceResponse::Output& output, const uint32_t class_index,
+    const char** label) const
+{
+  const auto& label_provider = backend_->GetLabelProvider();
```

Comment:
So in the case of ensemble, it will need to store the label provider from other backends internally so that here it can retrieve the one corresponding to the output?

Reply:
When initializing the LabelProvider for an ensemble backend, you should look at the ensemble outputs and find the corresponding labels attached to that output (from other models) and then load those into the ensemble's label provider.